core/src/main/java/org/apache/iceberg/ManifestEntry.java

```diff
@@ -25,7 +25,7 @@
 import static org.apache.iceberg.types.Types.NestedField.optional;
 import static org.apache.iceberg.types.Types.NestedField.required;
 
-interface ManifestEntry<F extends ContentFile<F>> {
+public interface ManifestEntry<F extends ContentFile<F>> {
```

Comment:
We could, if it was more palatable, expose a higher level api such as getAddedFiles, getExistingFiles, .... and so on so that callers don't need to know this lower level detail. I'm not sure if that is better since you still basically need to expose the enum values anyway ...

Reply:
@rdblue had an idea of building `ManifestExpirationManager` similarly to `ManifestMergeManager` and `ManfiestFilterManager` to encapsulate the logic of finding files to delete. The utility class will accept manifests to scan. Then we can either parallelize the scanning using Spark or run that locally. That should be a clean and unified way of handling this.

Reply:
Sounds good to me, This is what I was thinking about with that higher level api exposing. So the end user doesn't see anything with the inner enums. I'll get on this asap!